Dart/src/com/jetbrains/lang/dart/validation/fixes/DartServerFixIntention.java

```diff
@@ -180,9 +184,23 @@ public boolean startInWriteAction() {
   @Override
   public void invoke(@NotNull final Project project, final Editor editor, final PsiFile file) throws IncorrectOperationException {
     final SourceFileEdit fileEdit = myChange.getEdits().get(0);
+    final String filePath = fileEdit.getFile();
     final SourceEdit sourceEdit = fileEdit.getEdits().get(0);
 
-    final VirtualFile virtualFile = LocalFileSystem.getInstance().findFileByPath(FileUtil.toSystemIndependentName(fileEdit.getFile()));
+    // Create the file if it does not exist.
+    if (fileEdit.getFileStamp() == -1) {
+      try {
+        final File ioFile = new File(filePath);
```

Comment:
Do not use java.io.File when working with files that belong to the project. Such change is treated as external, undo doesn't work. Use VfsUtil.createDirectoryIfMissing and Virtualfile.createChildData().